src/main/scala/com/kaizen/api/actor/repository/InMemoryActorRepository.scala

```diff
@@ -0,0 +1,30 @@
+package com.kaizen.api.actor.repository
+
+import com.kaizen.api.RepositoryError
+import com.kaizen.api.actor.{Actor, ActorId}
+import zio.stm.TMap
+import zquery.DataSource
+
+class InMemoryActorRepository(memory: TMap[ActorId, Actor]) extends ActorRepository.Impl {
+  override val getById: DataSource[Any, GetActorById] =
+    DataSource.fromFunctionM("getActorById") { getActorById =>
+      memory
+        .get(getActorById.id)
+        .commit
+        .someOrFail(RepositoryError.ItemNotFound(getActorById.id).asInstanceOf[RepositoryError])
```

Comment:
I think if you annotate this with `RepositoryError`, you shouldn't need to cast it